tests/integration/profile_test.py

```diff
@@ -0,0 +1,125 @@
+#
+# Copyright (c) 2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+import os
+
+
+import pytest
+
+import libnmstate
+from libnmstate.schema import Interface
+from libnmstate.schema import InterfaceType
+from libnmstate.schema import InterfaceState
+
+from .testlib import cmd as libcmd
+
+
+DUMMY0_IFNAME = 'dummy0'
+
+NMCLI_CON_ADD_CMD = [
+    'nmcli',
+    'con',
+    'add',
+    'type',
+    'dummy',
+    'con-name',
+    'testProfile',
+    'connection.autoconnect',
+    'no',
+    'ifname',
+    DUMMY0_IFNAME,
+]
+
+PROFILE_DIRECTORY = '/etc/NetworkManager/system-connections/'
+
+
+def test_delete_new_interface_inactive_profiles(dummy_inactive_profile):
+    with dummy_interface(dummy_inactive_profile):
+        profile_exists = _profile_exists('testProfile.nmconnection')
+        assert not profile_exists
+
+
+def test_delete_existing_interface_inactive_profiles(dummy_inactive_profile):
+    with dummy_interface(dummy_inactive_profile):
+        _prepare_profiles_for_editting(DUMMY0_IFNAME)
+        desired_state = {
+            Interface.KEY: [
+                {
+                    Interface.NAME: DUMMY0_IFNAME,
+                    Interface.TYPE: InterfaceType.DUMMY,
+                    Interface.STATE: InterfaceState.UP,
+                    Interface.MTU: 2000,
+                }
+            ]
+        }
+        libnmstate.apply(desired_state)
+        profile_exists = _profile_exists('testProfile.nmconnection')
+        assert not profile_exists
+
+
+@contextmanager
+def dummy_interface(ifname):
+    try:
+        desired_state = {
+            Interface.KEY: [
+                {
+                    Interface.NAME: ifname,
+                    Interface.TYPE: InterfaceType.DUMMY,
+                    Interface.STATE: InterfaceState.UP,
+                }
+            ]
+        }
+        libnmstate.apply(desired_state)
+        yield
+    finally:
+        dummy0_dstate = desired_state[Interface.KEY][0]
+        dummy0_dstate[Interface.STATE] = InterfaceState.ABSENT
+        libnmstate.apply(desired_state)
+
+
+@pytest.fixture
+def dummy_inactive_profile():
+    try:
```

Comment:
The try is either after the exec_cmd or just before the yield.
(if the command to create fails, it means there is nothing there, so there is nothing to clean)

Reply:
Done, thanks!